users/serializers.py

```diff
@@ -10,3 +22,72 @@ def get_token(cls: Any, user: Any) -> Any:
         token["userID"] = user.id
         token["editor"] = user.is_editor
         return token
+
+
+class PasswordResetRequestSerializer(serializers.Serializer):
+    email = serializers.EmailField(required=True)
+
+    class Meta:
+        fields = ["email"]
+
+    def validate(self, attrs: Any) -> Any:
+        try:
+            request = self.context.get("request")
+            email = attrs.get("email")
+            if User.objects.filter(email=email).exists():
+                user = User.objects.get(email=email)
+                uidb64 = urlsafe_base64_encode(smart_bytes(user.id))
+                token = PasswordResetTokenGenerator().make_token(user)
+
+                # create email
+                current_site = get_current_site(request).domain
+                relative_link = reverse(
+                    "reset-password",
+                    kwargs={"uidb64": uidb64, "token": token},
+                )
+                absurl = f"http://{current_site}{relative_link}"
+                body = {"user": user, "link": absurl}
+                data = {
+                    "subject": "PASSWORD RESET",
+                    "body": body,
+                    "recipient": user.email,
+                }
+                Util.send_email(data)
+        except KeyError:
+            return Response(
+                {"error": "KeyError"}, status=status.HTTP_400_BAD_REQUEST
+            )
+        return super().validate(attrs)
+
+
+class PasswordResetSerializer(serializers.Serializer):
+    password = serializers.CharField(required=True, min_length=6)
+    uidb64 = serializers.CharField(required=True)
+    token = serializers.CharField(required=True)
+
+    class Meta:
+        fields = ["password", "uidb64", "token"]
+
+    def validate(self, attrs: Any) -> Any:
+        try:
+            password = attrs.get("password")
+            uidb64 = attrs.get("uidb64")
+            token = attrs.get("token")
+            user_id = smart_str(urlsafe_base64_decode(uidb64))
+            user = User.objects.get(id=user_id)
+
+            if not PasswordResetTokenGenerator().check_token(user, token):
+                return Response(
+                    {"message": "request failed"},
+                    status=status.HTTP_401_UNAUTHORIZED,
+                )
+
+            user.set_password(password)
+            user.save()
+
+        except Exception:
+            return Response(
```

Comment:
Do not return a response from the serializer, yyou can raise a PermissionDenied from rest_framework.exceptions